src/data/autogenerated-forms-configurator/DataStoreConfigurationRepository.ts

```diff
@@ -46,11 +47,127 @@ export class Dhis2DataStoreConfigurationRepository implements DataStoreConfigura
             }));
     }
 
-    async getConstants(): Promise<string[]> {
-        const { constants } = await this.api
-            .get<{ constants: { code: string }[] }>("/constants?fields=id,code")
+    private async getMetadata() {
+        const { objects: dataSets } = await this.api.models.dataSets
+            .get({
+                fields: dataSetFields,
+                filter: {
+                    name: {
+                        ilike: "NHWA",
+                        like: "MAL",
+                    },
```

Comment:
This filter means the app only works with these dataSets? (if that's a requirement is fine, just wondering since we're trying to do a more generic app)

Reply:
yes. i used this filter as i assumed we would only be working with MAL and NHWA datasets. could @MiquelAdell confirm please?

Reply:
@deeonwuli right now we are only working with these datasets but the idea is to be as generic is possible.
Why is this needed?

Reply:
it is not needed then. thank you @MiquelAdell